File: src/main/java/ch/obermuhlner/astro/gradient/operation/SubtractLinearImageOperation.java
package ch.obermuhlner.astro.gradient.operation;

public class SubtractLinearImageOperation extends AbstractSimpleChannelImageOperation {

  @Override
  protected double channelOperation(double channel1, double channel2, int x, int y, int channelIndex) {
    return (channel1 - channel2) / (1.0 - channel2);
  }

  @Override
  public String toString() {
    return "SubtractLinear";
  }
}

File: src/main/java/ch/obermuhlner/astro/gradient/operation/SubtractLinearImageOperation.kt
package ch.obermuhlner.astro.gradient.operation

class SubtractLinearImageOperation : AbstractSimpleChannelImageOperation() {
    override fun channelOperation(channel1: Double, channel2: Double, x: Int, y: Int, channelIndex: Int): Double {
        return (channel1 - channel2) / (1.0 - channel2)
    }

    override fun toString(): String {
        return "SubtractLinear"
    }
}